src/averageAges.js

```diff
@@ -55,7 +73,43 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersArr = people
```

Comment:
this solution is too complicated, let's try this way
you need to find women with children or only sons and push their age to the array

```
const ageDiffs = [];

if (onlyWithSon) {
    people.forEach(woman => people.map(person => {
      if (condition to check of woman has son) {
        ageDiffs.push(person.born - woman.born);
      }
    }));
  } else {
 ....
}
```